bee-common/bee-common/src/packable.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+//! A module that provides a `Packable` trait to serialize and deserialize types.
+
+pub use std::io::{Read, Write};
+
+/// A trait to pack and unpack types to and from bytes.
+pub trait Packable {
+    /// Associated error type.
+    type Error;
+
+    /// Returns the length of the packed bytes.
+    fn packed_len(&self) -> usize;
```

Comment:
Perhaps this could be changed to be an approximation for the purpose of capacity allocation? That way, it might be made considerably less costly to compute in some cases (i.e: hashmaps not needing to iterate all elements) without being 'incorrect'.

Reply:
I am not sure we can do this. It's sometimes used for verification purpose to compare the length of something read with an announced length.

Reply:
Meaning there are situations we actually rely on this being the exact size.

Reply:
Okay, fair enough